feg/gateway/services/n7_n40_proxy/n7/config.go

```diff
@@ -0,0 +1,136 @@
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package n7
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/golang/glog"
+
+	mcfgprotos "magma/feg/cloud/go/protos/mconfig"
+	"magma/feg/gateway/utils"
+	"magma/gateway/mconfig"
+)
+
+const (
+	N7N40ProxyServiceName = "n7_n40_proxy"
+
+	PcfApiRoot              = "PCF_API_ROOT"
+	PcfTokenUrl             = "PCF_TOKEN_URL"
+	PcfClientId             = "PCF_CLIENT_ID"
+	PcfClientSecret         = "PCF_CLIENT_SECRET"
+	N7ConsumerLocalAddr     = "N7_CONSUMER_LOCAL_ADDR"
+	N7ConsumerNotifyApiRoot = "N&_CONSUMER_NOTIFY_API_ROOT"
+
+	DefaultPcfApiRoot        = "https://localhost"
+	DefaultPcfTokenUrl       = "https://localhost/token"
+	DefaultClientId          = "magma_client_id"
+	DefaultClientSecret      = "magma_client_secret"
+	DefaultN7ConsumerAddr    = "localhost:0"
+	DefaultN7ConsumerApiRoot = "https://localhost/npcf-smpolicycontrol/v1"
+)
+
+type PCFConfig struct {
+	ApiRoot      string
+	TokenUrl     string
+	ClientId     string
+	ClientSecret string
+}
+
+type N7ConsumerConfig struct {
+	LocalAddr     string // ip:port or :port
+	NotifyApiRoot string
+}
+
+type N7Config struct {
+	DisableN7 bool
+	Server    *PCFConfig
+	Consumer  N7ConsumerConfig
+}
+
+func (cfg *PCFConfig) Validate() error {
+	_, err := url.ParseRequestURI(cfg.ApiRoot)
+	if err != nil {
+		return fmt.Errorf("invalid ApiRoot")
+	}
+	if len(cfg.TokenUrl) == 0 {
+		_, err = url.ParseRequestURI(cfg.TokenUrl)
+		if err != nil {
+			return fmt.Errorf("invalid TokenUrl")
+		}
+	}
+	return nil
+}
+
+func (cfg *N7ConsumerConfig) Validate() error {
+	_, err := url.ParseRequestURI(cfg.NotifyApiRoot)
+	if err != nil {
+		return fmt.Errorf("invalid NotifyApiRoot")
+	}
+	return nil
+}
+
+func GetN7Config() *N7Config {
+	configPtr := &mcfgprotos.N7N40ProxyConfig{}
+	conf := &N7Config{}
+
+	err := mconfig.GetServiceConfigs(N7N40ProxyServiceName, configPtr)
+	if err != nil || !validManagedConfig(configPtr) {
+		glog.V(2).Infof("Managed Configs Load Error: %v Using EnvVars", err)
+		conf.Server = &PCFConfig{
+			ApiRoot:      utils.GetValueOrEnv("", PcfApiRoot, DefaultPcfApiRoot),
+			TokenUrl:     utils.GetValueOrEnv("", PcfTokenUrl, DefaultPcfTokenUrl),
+			ClientId:     utils.GetValueOrEnv("", PcfClientId, DefaultClientId),
+			ClientSecret: utils.GetValueOrEnv("", PcfClientSecret, DefaultClientSecret),
+		}
+		conf.DisableN7 = false
+		conf.Consumer = N7ConsumerConfig{
+			LocalAddr:     utils.GetValueOrEnv("", N7ConsumerLocalAddr, DefaultN7ConsumerAddr),
+			NotifyApiRoot: utils.GetValueOrEnv("", N7ConsumerNotifyApiRoot, DefaultN7ConsumerApiRoot),
+		}
```

Comment:
this configuration is not validated. Lets do it at the end

Reply:
Done.